lib/bundler/resolver.rb

```diff
@@ -267,7 +268,9 @@ def search_for(dependency)
           []
         end
       end
-      search.select {|sg| sg.for?(platform, @ruby_version) }.each {|sg| sg.activate_platform!(platform) }
+      platform_results = search.select {|sg| sg.for?(platform, @ruby_version) }.each {|sg| sg.activate_platform!(platform) }
+      return platform_results if @gem_version_promoter.level == :major # default behavior
+      @gem_version_promoter.sort_versions(dependency, platform_results)
```

Comment:
yeah, i wasn't sure if you wanted that part in, too---- well, you couldn't with the platform bit. i'll take a look.